lib/rubocop/cop/style/hash_as_last_array_item.rb

```diff
@@ -40,7 +40,7 @@ def last_array_item?(node)
           parent = node.parent
           return false unless parent
 
-          parent.array_type? && parent.values.last == node
+          parent.array_type? && parent.children.last.object_id == node.object_id
```

Comment:
I think using `equal?` can be a bit simpler :-)

```suggestion
          parent.array_type? && parent.children.last.equal?(node)
```

Reply:
Agreed your suggestion is simpler.  I went ahead and committed.  Happy to squash if necessary.